src/ShallowWrapper.js

```diff
@@ -520,6 +521,29 @@ export default class ShallowWrapper {
   }
 
   /**
+   * Return a JSON representation of the node.
+   *
+   * NOTE: can only be called on a wrapper of a single node.
+   *
+   * @returns {Object}
+   */
+  json() {
+    return this.single('json', () => {
+      if (this.type() == null) {
+        return this.node;
+      }
+
+      const childrens = compact(this.children().map(c => c.json()));
+      return {
+        type: this.name(),
+        props: omit(this.props(), 'children'),
+        children: (childrens.length) ? childrens : null,
+        $$typeof: Symbol.for('react.test.json'),
+      };
```

Comment:
Is this format specified by `jest-snapshot`? If so, maybe we should call this `snapshot` instead of `json()` to make it clear that this method is specifically for snapshot tests?